File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    letters = ['A', 'E', 'I', 'O', 'U']
    count = 0
    answer = 0
    
    def dfs(current):
        nonlocal count , answer
        if current == word:
            answer = count
            return
        if len(current) == 5:
            return
        for ch in letters:
            count +=1
            dfs(current + ch)
    dfs('')
    return answer 

File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
class Solution{
    String[] vowels = {"A" , "E" , "I" , "O" , "U"};
    int count = 0;
    int answer = 0;
    String target;
    
    public int solution (String word){
        target = word;
        dfs("");
        return answer;
    }
    
    void dfs(String cur){
        if(!cur.equals("")){
            count++;
            
            if(cur.equals(target)){
                answer = count;
                return;
            }
        }
    
        if(cur.length() == 5) return;
    
        for(int i = 0; i < 5; i++){
            dfs(cur + vowels[i]);
            if(answer != 0) return;
        }
    }
}
